torch/distributed/nn/jit/instantiator.py

```diff
@@ -0,0 +1,135 @@
+
+    arg_str_list = []
+    arg_type_str_list = []
+    for argument in method_schema.arguments:
+        arg_str_list.append(argument.name)
+
+        if argument.has_default_value():
+            default_value_str = " = {}".format(argument.default)
+        else:
+            default_value_str = ""
+        arg_type_str = "{name}: {type}{default_value}".format(
+            name=argument.name, type=argument.type, default_value=default_value_str
+        )
+        arg_type_str_list.append(arg_type_str)
+
+    arg_str_list = arg_str_list[1:]  # Remove "self".
+    args_str = ", ".join(arg_str_list)
+
+    arg_type_str_list = arg_type_str_list[1:]  # Remove "self".
+    arg_types_str = ", ".join(arg_type_str_list)
+
+    assert len(method_schema.returns) == 1
+    argument = method_schema.returns[0]
+    return_type_str = str(argument.type)
+
+    return args_str, arg_types_str, return_type_str
+
+
+def write(out_path, text):
+    try:
+        with open(out_path, "r") as f:
+            old_text = f.read()
+    except IOError:
+        old_text = None
+    if old_text != text:
+        with open(out_path, "w") as f:
+            print("Writing {}".format(out_path))
+            f.write(text)
+    else:
+        print("Skipped writing {}".format(out_path))
+
+
+def instantiate_remote_module_template(
+    generated_module_name, module_interface_cls, is_scriptable
+):
+    if is_scriptable:
+        args_str, arg_types_str, return_type_str = get_arg_return_types_from_interface(
+            module_interface_cls
+        )
+        kwargs_str = ""
+        arrow_and_return_type_str = f" -> {return_type_str}"
+        arrow_and_future_return_type_str = f" -> Future[{return_type_str}]"
+    else:
+        args_str = "*args"
+        kwargs_str = "**kwargs"
+        arg_types_str = "*args, **kwargs"
+        arrow_and_return_type_str = ""
+        arrow_and_future_return_type_str = ""
+
+    jit_decorator_str_map = dict(
+        jit_script_decorator="@torch.jit.script",
+        jit_export_decorator="@torch.jit.export",
+    )
+    if is_scriptable is False:
+        jit_decorator_str_map = {
+            key: "" for key, value in jit_decorator_str_map.items()
+        }
+
+    remote_forward_template = CodeTemplate.from_file(
+        os.path.join(TEMPLATE_DIR_PATH, "remote_module.py.template")
+    )
+    env = dict(
+        generated_module_name=generated_module_name,
+        module_interface_cls_module_name=module_interface_cls.__module__,
+        module_interface_cls_name=module_interface_cls.__name__,
+        arg_types=arg_types_str,
+        arrow_and_return_type=arrow_and_return_type_str,
+        arrow_and_future_return_type=arrow_and_future_return_type_str,
+        args=args_str,
+        kwargs=kwargs_str,
+        **jit_decorator_str_map,
+    )
+    generated_code_text = remote_forward_template.substitute(env)
+    out_path = os.path.join(
+        TEMPLATE_DIR_PATH, "instantiated", f"{generated_module_name}.py"
```

Comment:
Question, we are putting the generated file to the src folder? Which can potentially be in Python packages? I am not aware about the convention here, but does it make more sense to put this to /tmp folder? Otherwise, the number of different modules can keep growing in the package folder?

Reply:
> Question, we are putting the generated file to the src folder? Which can potentially be in Python packages? I am not aware about the convention here, but does it make more sense to put this to /tmp folder?

We can only put them into Python packages (folders with `__init__.py`), so that we can import them, right?
If we put them into `/tmp`, we can't import them.

> Otherwise, the number of different modules can keep growing in the package folder?
Since 
1) the template directory path is known to template instantiator 
and 2) RemoteModule only works on top of rpc, 
I am thinking about add a helper function that purge the template instances folder, and call this helper in `rpc.shutdown()`.